bundles/io/org.eclipse.smarthome.io.transport.serial.api/src/main/java/org/eclipse/smarthome/io/transport/serial/api/SerialPortManager.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.transport.serial.api;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Interface for a serial port manager.
+ *
+ * @author Markus Rathgeb - Initial contribution
+ */
+@NonNullByDefault
+public interface SerialPortManager {
+    /**
+     * Gets an serial port identifier for a given name.
+     *
+     * @param name the name
+     * @return a serial port identifier or null
+     */
+    default @Nullable SerialPortIdentifier getIdentifier(final String name) {
+        final Optional<SerialPortIdentifier> opt = getIdentifiers().filter(id -> id.getName().equals(name)).findFirst();
+        if (opt.isPresent()) {
+            return opt.get();
+        } else {
+            return null;
+        }
+        // return getIdentifiers().filter(id -> id.getName().equals(name)).findFirst().orElse(null);
```

Comment:
can this be removed?

Reply:
Hm, I am not lucky about that.

There is a problem caused by our usage of the nullness annotations and the miss of EEAs.

The interface is marked as "non null by default".
The optional type is a generic one.
The type parameter `T` of `Optional<T>` is set to `SerialPortIdentifier` (the variable opt is of type `Optional<SerialPortIdentifier>`).
As types are now default non null, the tooling consider `T` as `@NonNull SerialPortIdentifier`.

The optional's method `orElse` looks like `public T orElse(T other)` (see [Optional API]).
Now, the tooling things the method looks like `public @NonNull SerialPortIdentifier orElse(@NonNullSerialPortIdentifier other)`.
But that is not correct: "other - the value to be returned if there is no value present, may be null".

So, the EEA needs to state, that regardless if T is nullable or not the method looks like:
```Java
public @NonNull T orElse(@Nullable T other);
```

So, instead of using the method that is part of the Java API and does exactly what we want, we have to write other code just to fulfill the tooling (or better "to workaround a wrong tooling behavior").

[Optional API]: https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElse-T-